Essentials/src/com/earth2me/essentials/User.java

```diff
@@ -220,7 +220,34 @@ public void takeMoney(final BigDecimal value, final CommandSource initiator, Use
             initiator.sendMessage(tl("takenFromOthersAccount", NumberUtil.displayCurrency(value, ess), this.getDisplayName(), NumberUtil.displayCurrency(getMoney(), ess)));
         }
     }
-
+    public void percentHandler(final BigDecimal percent, final CommandSource initiator, UserBalanceUpdateEvent.Cause cause, int cmd) {
+        String message = "PercentGivenTo";
+        if (percent.signum() == 0) {
+            return;
+        }
+        BigDecimal value = getMoney().multiply(percent.divide(new BigDecimal("100.0")));
+        try {
+            switch (cmd) {
```

Comment:
Noooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooope

Begone with your magic numbers!!!!

Make this an enum or refactor out - prefer the second to the first but :man_shrugging: